src/App.tsx

```diff
@@ -1,36 +1,56 @@
-import { Flex, useDisclosure, Text, Box } from '@chakra-ui/react'
+import { Box, Flex, Text, useDisclosure } from '@chakra-ui/react'
 import Overview from './views/Overview'
 import { ViewState } from './constants/enums'
 import { useApplicationContext } from './contexts/applicationContext'
 import Macroview from './views/Macroview'
 import { useEffect } from 'react'
-import SettingsModal from './views/SettingsModal'
 import data from '@emoji-mart/data'
 import { init } from 'emoji-mart'
 import './App.css'
 import { MacroProvider } from './contexts/macroContext'
+import MacroSettingsModal from './views/MacroSettingsModal'
+import SettingsModal from './views/SettingsModal'
 
 function App() {
-  const { viewState, initComplete } = useApplicationContext()
-  const { isOpen, onOpen, onClose } = useDisclosure()
+  const { viewState, initComplete, appDebugMode } = useApplicationContext()
+  const {
+    isOpen: isOpenSettings,
+    onOpen: onOpenSettings,
+    onClose: onCloseSettings
+  } = useDisclosure()
+  const {
+    isOpen: isOpenMacroSettings,
+    onOpen: onOpenMacroSettings,
+    onClose: onCloseMacroSettings
+  } = useDisclosure()
+
+  if (appDebugMode !== null && !appDebugMode) {
+    // Disables Tauri right click context menu
+    document.addEventListener('contextmenu', (event) => event.preventDefault())
+    // Ctrl + f is disabled with this event listener to whether the debug mode is on
+    document.addEventListener('keydown', (event) => {
+      if (event.ctrlKey && event.key.toLowerCase() === 'f') {
+        event.preventDefault()
+      }
+      if (event.ctrlKey && event.key.toLowerCase() === 'r') {
+        event.preventDefault()
+      }
+      if (event.key.toLowerCase() === 'f5') {
+        event.preventDefault()
+      }
+    })
+    document.addEventListener('selectstart', (event) => event.preventDefault())
+    console.log("prevented default action because debug is: ", appDebugMode)
+  }
```

Comment:
omg this code is so dangerous lol, it's going to add a new event listener every render. Ideally, you should maybe split this into its own hook, and add them in a useEffect, where the cleanup function of the useEffect will remove the listeners.

Reply:
Tried improving, something like this?: [10217c6](https://github.com/WootingKb/wooting-macros/pull/194/commits/10217c670081cd13157f81b0d20ed904c082bb4a)

Reply:
Tweaked more: [f9ba666](https://github.com/WootingKb/wooting-macros/pull/194/commits/f9ba6668f49d42c36e6ae1f742733dbfb327219f)